vm/src/hint_processor/builtin_hint_processor/bootloader/simple_bootloader_hints.rs

```diff
@@ -0,0 +1,171 @@
+    use std::collections::HashMap;
+
+    use felt::Felt252;
+    use rstest::{fixture, rstest};
+
+    use crate::hint_processor::builtin_hint_processor::bootloader::simple_bootloader_hints::{
+        prepare_task_range_checks, set_ap_to_zero,
+    };
+    use crate::hint_processor::builtin_hint_processor::bootloader::types::{
+        FactTopology, SimpleBootloaderInput, Task,
+    };
+    use crate::hint_processor::builtin_hint_processor::bootloader::vars;
+    use crate::hint_processor::builtin_hint_processor::hint_utils::get_ptr_from_var_name;
+    use crate::hint_processor::hint_processor_definition::HintReference;
+    use crate::serde::deserialize_program::ApTracking;
+    use crate::types::exec_scope::ExecutionScopes;
+    use crate::types::relocatable::Relocatable;
+    use crate::utils::test_utils::*;
+    use crate::vm::vm_core::VirtualMachine;
+
+    #[fixture]
+    fn simple_bootloader_input() -> SimpleBootloaderInput {
+        SimpleBootloaderInput {
+            fact_topologies_path: None,
+            single_page: false,
+            tasks: vec![Task {}, Task {}],
+        }
+    }
+
+    #[rstest]
+    fn test_prepare_task_range_checks(simple_bootloader_input: SimpleBootloaderInput) {
+        let mut vm = vm!();
+        vm.run_context.fp = 3;
+        vm.segments = segments![((1, 0), (2, 0)), ((1, 1), (2, 2))];
+        let ids_data = ids_data!["output_ptr", "range_check_ptr", "task_range_check_ptr"];
+        vm.add_memory_segment();
+
+        let mut exec_scopes = ExecutionScopes::new();
+        exec_scopes.insert_value(
+            vars::SIMPLE_BOOTLOADER_INPUT,
+            simple_bootloader_input.clone(),
+        );
+
+        let ap_tracking = ApTracking::new();
+
+        prepare_task_range_checks(&mut vm, &mut exec_scopes, &ids_data, &ap_tracking)
+            .expect("Hint failed unexpectedly");
+
+        let task_range_check_ptr =
+            get_ptr_from_var_name("task_range_check_ptr", &mut vm, &ids_data, &ap_tracking)
+                .unwrap();
+
+        // Assert *output_ptr == n_tasks
+        let output = vm
+            .segments
+            .memory
+            .get_integer(Relocatable {
+                segment_index: 2,
+                offset: 0,
+            })
+            .unwrap()
+            .to_usize()
+            .unwrap();
+        assert_eq!(output, simple_bootloader_input.tasks.len());
+
+        // Assert task_range_check_ptr == range_check_ptr (2, 2) + BUILTIN_DATA_SIZE (8) * n_tasks (2)
+        assert_eq!(
+            task_range_check_ptr,
+            Relocatable {
+                segment_index: 2,
+                offset: 18
+            }
+        );
+
+        let fact_topologies: Vec<FactTopology> = exec_scopes
+            .get(vars::FACT_TOPOLOGIES)
+            .expect("Fact topologies missing from scope");
+        assert!(fact_topologies.is_empty());
+    }
+
+    #[rstest]
+    fn test_set_to_zero() {
+        let mut vm = vm!();
+        add_segments!(vm, 2);
+
```

Comment:
Nit: testing that segments is non-zero here would be more thorough, although that may seem extremely mundane to someone who understands the segments better :)

Reply:
In Cairo, memory is only writable once. If I try to read at the address I will get an error back (unallocated memory error, iirc).